dependency/retrieval/main.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"flag"
+	"log"
+	"os"
+
+	"github.com/paketo-buildpacks/dotnet-core-sdk/dependency/retrieval/components"
+	"github.com/paketo-buildpacks/packit/v2/cargo"
+)
+
+func main() {
+	var buildpackTOMLPath, outputPath string
+	set := flag.NewFlagSet("", flag.ContinueOnError)
+	set.StringVar(&buildpackTOMLPath, "buildpack-toml-path", "", "path to the buildpack.toml file")
+	set.StringVar(&outputPath, "output", "", "path to the output file")
+	err := set.Parse(os.Args[1:])
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fetcher := components.NewFetcher()
+	releases, err := fetcher.Get()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var versions []string
+	for _, release := range releases {
+		versions = append(versions, release.SemVer.String())
+	}
+
+	newVersions, err := components.FindNewVersions(buildpackTOMLPath, versions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var dependencies []cargo.ConfigMetadataDependency
+	for _, version := range newVersions {
+		for _, r := range releases {
```

Comment:
its sort of unfortunate that you have to loop over EVERY release of .NET in existence to find the matching one for a new version. Maybe `releases` as a `map[string]Release` mapping the version (as a string) to the release, would be easier to work?

Reply:
See above explanation.

Reply:
Now that the number of dependency from `FindNewVersions` is reduced I think that this is a bit less of a concern even more.